src/test/compile-fail/or-patterns.rs

```diff
@@ -0,0 +1,56 @@
+// should-ice
```

Comment:
Can you move the test to `src/test/ui/or-patterns/`?

Reply:
The test harness refuses `should-ice` outside of `compile-fail`, for some reason. I would have preferred it to be an `ui` test too. We'll need to move it once it stops ICEing

Reply:
Hmm; could we change `matches.rs` to emit an error and `abort_if_errors` temporarily instead of calling `self.error_simplifyable` in case of `PatKind::Or { .. } =>`?